src/AWS.Deploy.Orchestration/DisplayedResources/DisplayedResourcesHandler.cs

```diff
@@ -0,0 +1,54 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using AWS.Deploy.Common;
+using System.Threading.Tasks;
+using AWS.Deploy.Orchestration.Data;
+
+namespace AWS.Deploy.Orchestration.DisplayedResources
+{
+    public interface IDisplayedResourcesHandler
+    {
+        Task<List<DisplayedResourceItem>> GetDeploymentOutputs(CloudApplication cloudApplication, Recommendation recommendation);
+    }
+
+    public class DisplayedResourcesHandler : IDisplayedResourcesHandler
+    {
+        private readonly IAWSResourceQueryer _awsResourceQueryer;
+        private readonly IDisplayedResourceCommandFactory _displayedResourcesFactory;
+
+        public DisplayedResourcesHandler(IAWSResourceQueryer awsResourceQueryer, IDisplayedResourceCommandFactory displayedResourcesFactory)
+        {
+            _awsResourceQueryer = awsResourceQueryer;
+            _displayedResourcesFactory = displayedResourcesFactory;
+        }
+
+        /// <summary>
+        /// Retrieves the displayed resource data for known resource types by executing specific resource commands.
+        /// For unknown resource types, this returns the physical resource ID and type.
+        /// </summary>
+        public async Task<List<DisplayedResourceItem>> GetDeploymentOutputs(CloudApplication cloudApplication, Recommendation recommendation)
+        {
+            var displayedResources = new List<DisplayedResourceItem>();
+
+            if (recommendation.Recipe.DisplayedResources == null)
+                return displayedResources;
+
+            foreach (var displayedResource in recommendation.Recipe.DisplayedResources)
+            {
+                var resource = await _awsResourceQueryer.DescribeCloudFormationResource(cloudApplication.StackName, displayedResource.LogicalId);
```

Comment:
I think we should just make once call to `DescribeStackResources` to get all of the resources instead of making individual call for each resources. Seems like we average 3 displayed resources per recipe which means roughly 3 times the latency for calling CloudFormation and we run more risk of getting throttled by CF.

Reply:
Done